docs/en/styling-and-theming/supplemental.md

```diff
@@ -0,0 +1,642 @@
+    ```
+
+3.  Import the theme typescript module and use it [as per any other theme](#making-themeable-applications):
+
+    ```ts
+    import theme from '@dojo/themes/dojo';
+
+    render() {
+    	return w(Button, { theme }, [ 'Hello World' ]);
+    }
+    ```
+
+If attempting to use the themes in custom elements, after installing `@dojo/themes`:
+
+1.  Add the custom element-specific theme CSS to `index.html`:
+
+    ```html
+    <link rel="stylesheet" href="node_modules/@dojo/themes/dojo/dojo-{version}.css" />
+    ```
+
+2.  Add the custom element-specific theme JS to `index.html`:
+
+    ```html
+    <script src="node_modules/@dojo/themes/dojo/dojo-{version}.js"></script>
+    ```
+
+### Composing off Dojo themes
+
+Once `@dojo/themes` is installed in a project, it can be used as the basis for an extended application theme by including relevant components with [CSS modules' composes functionality](#css-module-composition) in the new theme.
+
+`@dojo/themes` also includes its own [`:root` `variables.css` file](#css-custom-properties) which can be imported if the extended application theme would like to reference Dojo-specified properties elsewhere in the new theme.
+
+The following is an example of a new theme for the `@dojo/widgets/button` widget that extends off `@dojo/themes`, and changes the button's background to green while retaining all other button theme style properties:
+
+> src/themes/myTheme/theme.ts
+
+```ts
+import * as myButton from './myButton.m.css';
+
+export default {
+	'@dojo/widgets/button': myButton
+};
+```
+
+> src/themes/myTheme/myButton.m.css
+
+```css
+@import '@dojo/themes/dojo/variables.css';
+
+.root {
+	composes: root from '@dojo/themes/dojo/button.m.css';
+	background-color: var(--dojo-green);
+}
+```
+
+# Theming a Dojo application
+
+Dojo applications need a way to present all the widgets they use in a consistent manner, so that users perceive and interact with application features holistically, rather than as a mashup of disjointed elements on a webpage. This is usually implemented via a corporate or product marketing theme that specifies colors, layout, font families, and more.
+
+## Making themeable widgets
+
+There are three requirements for widgets to be considered themeable:
+
+1.  The widget's class should have the `@theme()` decorator applied, passing in the widget's imported CSS module as a decorator argument.
+2.  The widget's parent class should be wrapped with `ThemedMixin`.
+3.  One or more of the widget's styling classes should be wrapped in a call to `this.theme()` when rendering the widget.
+
+By convention, there is a fourth requirement that is useful when developing widgets intended for distribution (this is a convention that widgets in Dojo's widget library follow):
+
+4.  The widget's main styling class should be named `root`. Doing so provides a predictable way to target the main styling class of any third-party themeable widgets when overriding styles in a custom theme.
```

Comment:
Do we ever describe that the `root` class should be used on the parent node returned by the widget? Or at least is the general pattern widgets does.